Dart/src/com/jetbrains/lang/dart/ide/runner/server/vmService/DartVmServiceDebugProcess.java

```diff
@@ -279,13 +286,13 @@ public void guessRemoteProjectRoot(@NotNull final ElementList<LibraryRef> librar
 
         LOG.assertTrue(file.getPath().startsWith(localProjectRoot.getPath() + "/"), file.getPath() + "," + localProjectRoot.getPath());
         final String relPath = file.getPath().substring(localProjectRoot.getPath().length()); // starts with slash
-        if (!relPath.startsWith("/lib/") && remoteUri.endsWith(relPath)) {
+        if (relPath.startsWith("/lib/") && remoteUri.endsWith(relPath)) {
```

Comment:
I'm afraid I do not understand this change. For standard project this condition is always `false` because files from 'lib' folder do not have 'lib' in uri, they have 'package:' uri and can't help us to guess remote project root. 
My idea was to catch some file from, for example, 'bin' folder having URI=file:///remote/path/to/project/bin/foo/bar/file.dart, find local file in /local/path/to/project/bin/foo/bar/file.dart and make a conclusion that remote root is /remote/path/to/project.